src/probe.rs

```diff
@@ -115,12 +115,25 @@ impl ParseOptions {
 }
 
 /// The parsing strictness mode
-#[derive(Copy, Clone, Debug)]
+#[derive(Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Default)]
 #[non_exhaustive]
 pub enum ParsingMode {
-	/// The default mode, will eagerly error on invalid input
+	/// Will eagerly error on invalid input
+	///
+	/// This mode will eagerly error on any non spec-compliant input.
 	Strict,
-	/// Less eager to error, may produce invalid output
+	/// Default mode, less eager to error on recoverably malformed input
+	///
+	/// This mode will attempt to fill in any holes where possible in otherwise valid, spec-compliant input.
+	///
+	/// NOTE: A readable input does *not* necessarily make it writeable.
+	#[default]
+	BestAttempt,
+	/// Least eager to error, may produce invalid output
+	///
+	/// This mode will discard any invalid fields, and ignore the majority of non-fatal errors.
+	///
+	/// Tags and properties read using this mode may come out missing items if the input is malformed.
```

Comment:
The wording sounds complicated to me as a non-natice speaker. One might overlook that the final result could be incomplete if errors occurred.  It would help to explicitly use the terms "imcomplete" and/or "partial" in the wording.

Reply:
Good call :+1: